src/main/java/org/folio/rest/impl/PurchaseOrderLineHelper.java

```diff
@@ -333,6 +341,49 @@ private String buildPoLineNumber(String poNumber, String sequence) {
     return poNumber + "-" + sequence;
   }
 
+  /**
+   * See MODORDERS-180 for more details.
+   * @param compPoLine composite PO Line
+   */
+  private void updateEstimatedPrice(CompositePoLine compPoLine) {
+    Cost cost = compPoLine.getCost();
+    if (cost != null) {
+      cost.setPoLineEstimatedPrice(calculateEstimatedPrice(cost));
+    }
+  }
+
+  /**
+   * Calculates total estimated price. See MODORDERS-180 for more details.
+   * @param cost PO Line's cost
+   */
+  double calculateEstimatedPrice(Cost cost) {
+    BigDecimal total = BigDecimal.ZERO;
+    if (cost.getListUnitPrice() != null) {
+      BigDecimal pPrice = BigDecimal.valueOf(cost.getListUnitPrice())
+                                    .multiply(BigDecimal.valueOf(cost.getQuantityPhysical()));
+      total = total.add(pPrice);
+    }
+    if (cost.getListUnitPriceElectronic() != null) {
+      BigDecimal ePrice = BigDecimal.valueOf(cost.getListUnitPriceElectronic())
+                                    .multiply(BigDecimal.valueOf(cost.getQuantityElectronic()));
+      total = total.add(ePrice);
+    }
+    if (cost.getDiscount() != null) {
+      BigDecimal discount;
+      if (Cost.DiscountType.AMOUNT == cost.getDiscountType()) {
+        discount = BigDecimal.valueOf(cost.getDiscount());
+      } else {
+        discount = total.multiply(BigDecimal.valueOf(cost.getDiscount()/100d));
```

Comment:
The calculation in the story for discount amount is given as 
>discountAmount = discountType == PERCENTAGE ? baseListPrice * discount : discount. 
but here we are calculating per 100. Both are not wrong, but it comes down if cost.discount = 10 vs cost.discount= 0.1
Either way, this must be corrected or the story description(because we are referencing the story# in the method description we must make sure story descriptions are always correct)

Reply:
I had assumed discount would be a floating point value < 1 if discountType == PERCENT

Reply:
I think we might want to actually put some validation in place for this... something along the lines of if it's a percent and > 1 throw an error.

Reply:
Honestly thought I didn't catch this when reviewing the acq-models PR which IIRC has examples with discount=10, discountType=Percent, so good catch @KVupp.

Reply:
I think keeping this as-is is probably fine, but we need to make sure it's documented in the schema

Reply:
The description of the discount property in the schema file has been updated:
`Percentage (0 to 100) or amount (positive number) that is subtracted from the list price time quantities calculation before additional cost`